app/controllers/users_controller.rb

```diff
@@ -112,6 +111,9 @@ def show
     @phenotype_comment_replies.sort! { |b,a| a.created_at <=> b.created_at }
     @paginated_phenotype_replies = @phenotype_comment_replies
 
+    # get the last 30 updated SNPs for this user
+    @last_updated_snps = @user.snps.order('updated_at desc').limit(30) unless @user.snps.nil?
```

Comment:
It's quite a bit faster :) it runs a whole bunch of sql statements but each only takes ~5ms

```
Benchmark.measure {        @last_30_papers = Set.new
    if !@user.genotypes.nil?
      # get the last 30 updated SNPs for this user
      # we do it the other way around - check the last 20 updated papers for each type, see if their snps are in user's collection
      # the upper if is important since without it even users without SNPs will trigger this
      %w(Snpedia Mendeley GenomeGov Plos).each do |source|
        "#{source}Paper".constantize.last(30).each do |p|
          p.snps.each do |s|
            @last_30_papers.add(p) if s.users.include? @user
          end
        end
      end
      # convert to array, sort by updated_at, get newest update first
      @last_30_papers = @last_30_papers.to_a.sort_by(&:updated_at).reverse
      # in the best case we have 4*30 papers for this user now, in the worst case we have 0
      @last_30_papers = @last_30_papers[0..30]
    end

}
```
on the server with a two different users returns

    => #<Benchmark::Tms:0x00000006857428 @label="", @real=3.512849297374487, @cstime=0.0, @cutime=0.0, @stime=0.04999999999999982, @utime=0.6900000000000004, @total=0.7400000000000002>

This is a run with a randomly chosen user with uploaded genotype - 

With User.find(1) `@real` goes down to 1.6s, probably because it cached stuff from me playing around. 

But having a worker for that table would probably be the most sane way - have a worker that makes a list of 4*30 last updated workers (or have all API workers report their last updated ones? sounds dodgy), so that the user controller can just check whether current_user has any connection to that list.